test/client/feature/geojson-external-instance.spec.js

```diff
@@ -0,0 +1,211 @@
+import { geoJSONExternalInstance } from '../../../public/js/src/module/geojson';
+
+describe('GeoJSON external secondary instances', () => {
+    describe('validation failures', () => {
+        const failures = [
+            {
+                reason: 'if the data is not an object',
+                source: 'not-object.geojson',
+            },
+            {
+                reason: 'if the top-level type is not FeatureObject',
+                source: 'invalid-type.geojson',
+            },
+            {
+                reason: 'if it has no features property',
+                source: 'bad-futures-collection.geojson',
+            },
+            {
+                reason: 'if the features property is not an array',
+                source: 'bad-features-not-array.geojson',
+            },
+            {
+                reason: 'if an item in the features array does not have a type of Feature',
+                source: 'bad-feature-not-feature.geojson',
+            },
+            {
+                reason: 'if the geometry is not a point',
+                source: 'feature-collection-with-line.geojson',
+            },
+        ];
+
+        failures.forEach(({ reason, source }) => {
+            it(`fails to parse ${reason}`, async () => {
+                const url = `/base/test/fixtures/geojson/${source}`;
+                const response = await fetch(url);
+                const data = await response.json();
+
+                expect(
+                    geoJSONExternalInstance.bind(
+                        null,
+                        source.replace('.geojson', ''),
+                        data
+                    )
+                ).to.throw(TypeError);
+            });
+        });
+    });
+
+    it('ignoresExtraTopLevelProperties', async () => {
```

Comment:
Oops 🙃

This test suite is derived entirely from the [JavaRosa tests](https://github.com/lognaturel/javarosa/blob/e5925939df4ca5c72a03a435e8f3ee94c90c4bd4/src/test/java/org/javarosa/core/model/instance/geojson/GeoJsonExternalInstanceTest.java). I intended for that to be obvious, but didn't mean to make it _this_ obvious.